scripts/hack/release-helper.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+set -o pipefail
+
+this=`basename $0`
+
+usage () {
+cat << EOF
+Usage: $this VERSION
+
+Example:
+
+  $this v0.2.0
+EOF
+}
+
+# Check args
+if [ $# -ne 1 ]; then
+    usage
+    exit 1
+fi
+
+version=$1
+shift 1
+
+container_image=ghcr.io/containers/nri-plugins/nri-plugins-operator:$version
+
+if ! [[ $version =~ ^v[0-9]+\.[0-9]+\..+$ ]]; then
+    echo -e "ERROR: invalid VERSION '$version'"
+    exit 1
+fi
+
+# Patch Helm charts
+echo "Patching Helm charts"
+find ../../deployment/helm -name Chart.yaml | xargs -I '{}' \
+    sed -e s"/appVersion:.*/appVersion: $version/" -i '{}'
+find ../../deployment/helm -name values.yaml | xargs -I '{}' \
+    sed -e s"/pullPolicy:.*/pullPolicy: IfNotPresent/" -i '{}'
+
+# Patch deployment templates
+echo Patching kustomize templates to use $container_image
+make -C ../../deployment/operator kustomizations && \
+    find ../../deployment/operator/config/manager -name kustomization.yaml | xargs -I '{}' \
```

Comment:
with the current state, when this script is executed, the outcome would be the all charts version updated + operator `kustomization.yaml` file and `nri-plugins-operator.clusterserviceversion.yaml` created. So we don't expect `kustomization.yaml` and `nri-plugins-operator.clusterserviceversion.yaml` to be tracked on git. 

example:

```
fmuyassarov@dev:~/work/github/nri-plugins/scripts/hack$ ./release-helper.sh v0.1.1
Patching Helm charts
Patching kustomize templates to use ghcr.io/containers/nri-plugins/nri-plugins-operator:v0.1.1
make: Entering directory '/home/fmuyassarov/work/github/nri-plugins/deployment/operator'
make: Leaving directory '/home/fmuyassarov/work/github/nri-plugins/deployment/operator'
fmuyassarov@dev:~/work/github/nri-plugins/scripts/hack$ git status
On branch fix-image-ref-main
Your branch is up to date with 'origin/fix-image-ref-main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ../../deployment/helm/balloons/Chart.yaml
	modified:   ../../deployment/helm/balloons/values.yaml
	modified:   ../../deployment/helm/memory-qos/Chart.yaml
	modified:   ../../deployment/helm/memory-qos/values.yaml
	modified:   ../../deployment/helm/memtierd/Chart.yaml
	modified:   ../../deployment/helm/memtierd/values.yaml
	modified:   ../../deployment/helm/sgx-epc/Chart.yaml
	modified:   ../../deployment/helm/sgx-epc/values.yaml
	modified:   ../../deployment/helm/template/Chart.yaml
	modified:   ../../deployment/helm/template/values.yaml
	modified:   ../../deployment/helm/topology-aware/Chart.yaml
	modified:   ../../deployment/helm/topology-aware/values.yaml

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	../../deployment/operator/config/manager/kustomization.yaml
	../../deployment/operator/config/manifests/bases/nri-plugins-operator.clusterserviceversion.yaml

no changes added to commit (use "git add" and/or "git commit -a")

```

Reply:
But I think `make bundle` will patch those yamls anyway so there is not use editing them. Right? And no use patching untracked files in this script(?)

Reply:
You are right and now after you asked, I think we have it a bit incorrectly. So, there are two things we want to achieve:
1. Have our charts and operators manifests with unstable tag on the git (not a specific tag)
2. Automate the charts and operator manifests updating so that we don't have the run this script and submit changes every time before creating a new release. Because there is a change we might forget to run it. But then if we were to automate it, we would have kustomization.yaml and nri-plugins-operator.clusterserviceversion.yaml files generated on workflow environment and we can't submit them as part of bundle submission to artifacthub.io nor we can push them to the git. So, they are useless like you said. 

I think the best would be that, when those files are generated we also run the `make bundle-build` and then submit the operator new version a new pull request via bot.

Reply:
if we are anyways opting in for running release-helper.sh script manually before cutting a new release, then I don't see a point of having `kustomization.yaml.in` and `nri-plugins-operator.clusterserviceversion.yaml.in`

Reply:
should we switch back to regular yaml files instead @marquiz

Reply:
> should we switch back to regular yaml files instead @marquiz

@fmuyassarov I'm (probably) fine with anything reasonably maintainable, as long as I don't end up, as a side-effect, with uncommitted changes or an extra commit every single time in my normal workflow I do make, make images, make PLUGINS=... images, etc.